### ISSUE TITLE
report cursor position also in midi editor

### ISSUE THREAD

**Issue by ivandalia** [jcsteh/osara#57]
Hello James, 
I am Ivan Dalia, a student of Gianluca Apollaro, also from Italy.
I noticed that the action osara: report edit/play cursor position works only in the track view, would it be possible to make it work also within the midi view?
thancks alot

**Comment by jcsteh** [comment]
Do you mean the Event List or the Piano Role? I don't think it's not 
that useful in the latter, as the rest of the Piano Role isn't 
accessible yet. I can see about the Event List, though.

**Comment by Alex-Longard** [comment]
Hi James, on midi editor i use event list, in event property dialog i can not edit position and lenght the note and cc events.
This bug on osara or reaper?

And not correct work on hotkeys for move notes and events in event list.

**Comment by jcsteh** [comment]
> Hi James, on midi editor i use event list, in event property dialog i can not edit position and lenght the note and cc events.

Works fine for me, except CC events don't have a length, only a position.

> And not correct work on hotkeys for move notes and events in event list.

I don't understand the issue. Can you please rephrase? Note that any additional issues should be filed as separate issue reports; this is just for the report cursor position action.

**Comment by GianlucaApollaro** [comment]
Hi James:
Using the keymap Derek and I created, the piano Role is accessible: in fact I chained some actions and created some macros that allow us to preview and navigate note by note with up or down arrow, make operations on the selected notes like make them longer or shorter, increase or decrease their velocity. Here are some actions that I mapped and created included in our keymap:
Ctrl+Shift+Numbers in the number row set the grid
Numbers in the number row set the grid
Period key Custom: Set grid to dotted notes 
Ctrl+Alt+I fit notes to time selection
Up arrow Custom: Select and move cursor to previous note
Down arrow Custom: Select and move cursor to next note
2 (Tn) Move notes down one semitone
8 (Tn) Move notes up one semitone
Alt+2 (Tn) Move notes down one octave
Alt+8 (Tn) Move notes up one octave
4 (Tn) Move notes left one grid unit 
6 (Tn) Move notes right one grid unit 
J Join notes 
Delete key Delete events 
Left arrow Move edit cursor left by grid
Right arrow Move edit cursor right by grid
3 (Tn) Lengthen notes one grid unit 
1 (Tn) Shorten notes one grid unit 
9 (Tn) Note velocity +01 
Alt+9 (Tn) Note velocity +10 
7 (Tn) Note velocity -01 
Alt+7 (Tn) Note velocity -10 
Shift+Right Arrow Add next note to selection 
Shift+Left Arrow Add previous note to selection 
Alt+Shift+E Edit: Select all notes in time selection 
PageUp Move edit cursor back one measure
PageDown Move edit cursor forward one measure
Ctrl+F SWS/FNG: Select notes nearest edit cursor 
Ctrl+Shift+A Select all notes with same note value.
So I kind of see a utility in knowing where the cursor is

**Comment by Alex-Longard** [comment]
Hi James, Gian Luca ansver for me.
I thought the hot keys to move the notes in the event list
In my reaper 5.1 event properties not work correctly, i can not edit note position on editbox, this field is reset when press "ok" button.

**Comment by jcsteh** [comment]
Fair enough. Interestingly, the report cursor position action already 
seems to work in the Piano Role for me. It also kinda works in the Event 
List, but screen readers won't speak it because of the way focus works 
in lists; I'll have to tweak some things a bit.

**Comment by jcsteh** [comment]
I can edit note position and length here and the value stays the way I set it. If it's resetting, that suggests the format of the value is wrong or there's a bug in REAPER. In any case, this definitely has nothing to do with OSARA.

**Comment by GianlucaApollaro** [comment]
Hi,
I don't understand your question very much. All I can say is that I can 
edit position in the event list with no problem. I bring up the event 
list with Alt+3 from within the midi editor, select the note I want to 
edit and then I press CtrlF2 that brings up the event property, from 
there I can edit whatever value with no problem. You can get to the 
event property also from the piano roll after selecting a note.

Il 30/11/2015 12:47, Alex-Longard ha scritto:

**Comment by Alex-Longard** [comment]
In reaper 4.78 event properties work normal, in 5.1 i see bug as write up.
Gian Luca, thanks, good work on piano roll!

**Comment by Alex-Longard** [comment]
Hi all developers!
I can not move event "end" on event list, this event set on 2 measure and not moved to next measure.
What ideas you have?
My all problems related to this

**Comment by jcsteh** [comment]
This is off-topic for this issue, but in any case, it sounds like you're 
seeing an issue with REAPER, not with OSARA, so this isn't the right 
place for that answer. You could try asking on the rwp list, though.

**Comment by jcsteh** [comment]
Sorry; this completely broke JAWS. Back to square 0. This kind of silliness is why we can't have nice things.

**Comment by jcsteh** [comment]
After #198, OSARA can now report its messages in the MIDI Event List with screen readers that support UIA notification events (NVDA and Narrator), as well as VoiceOver on Mac. I can't fix this for other screen readers.
